website/static/js/share/results.js

```diff
@@ -37,79 +38,155 @@ Results.controller = function(vm) {
     self.vm = vm;
     self.vm.resultsLoading = m.prop(false);
 
+    self.renderTitleBar = function(result) {
```

Comment:
Good idea. I was just refactoring it in the style that it already was, I can definitely move everything to components as that seems much cleaner.